### PR TITLE
GpuDnnBatchNorm with 5d inputs

### PR DESCRIPTION
**Pull request by gvtulder**
Here's a little patch that would add 5d support to the GPU batch normalization ops (see #4931).

The tests pass now, but I'm not really sure about the tolerances used in the tests. Some tests would fail with what seemed to be relatively small differences. Using larger input values and changing some of the atol/rtol values fixed that.

### CODE REVIEW

File: theano/sandbox/cuda/dnn.py [hunk 1]
```diff
@@ -2395,23 +2392,22 @@ def c_code(self, node, name, inputs, outputs, sub):
...
-        result += c_set_tensor4d(scale, 'bn_params_' + name, 'err' + name, sub['fail'])
-
-        # build and prepare the output variable
+        # call cuDNN function
         result += """
-if (CudaNdarray_prep_output(&%(outp)s, 4, CudaNdarray_HOST_DIMS(%(inp)s)) != 0)
-{
+// set input tensor descriptors from input tensors
+if (c_set_tensorNd(%(inp)s, bn_input_%(name)s) != 0)
+    %(fail)s
```
Comment (nouiz) [inline theano/sandbox/cuda/dnn.py]: put fail between "{}". As it is hiden code, it could be more then 1 statement in the futur.

**Comment by nouiz** [comment]
jenkins test this

@abergeron can you review it? I think it is good, but you know that code more then me. What do you think about the threashold problem in tests?

File: theano/sandbox/cuda/dnn.py (hunk 1 above)
Comment (gvtulder) [inline theano/sandbox/cuda/dnn.py]: Done. (I copied this from elsewhere, by the way. fail without {} occurs in other places as well, so the future should be careful.)

**Comment by mdering** [comment]
Currently using this implementation with keras, and its working (and is a LOT faster). Thanks a lot!

**Comment by gvtulder** [comment]
The Jenkins build still failed, so I increased the rtol a little more.

On my system the tests passed with floatX=float32, but they would fail with floatX=float64 if I didn't increase the tolerance.

**Comment by mdering** [comment]
Will this be merged soon?

**Comment by gvtulder** [comment]
Rebased to the current master. (There was a conflict introduced by PR #5028).

**Comment by lamblin** [comment]
Jenkins, test this, please.

**Comment by gvtulder** [comment]
Jenkins passed.

**Comment by abergeron** [comment]
The gpuarray part is ok.  I didn't review the cuda part.  If we drop that part I would be ok with merging this.  Otherwise, someone else can review it.
